File: my-folder/1568-pseudo-palindromic-paths-in-a-binary-tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int pseudoPalindromicPaths(TreeNode root) {
        return dfs(root, new HashSet<>());
    }

    private int dfs(TreeNode root, Set<Integer> set) {
        if (root == null) {
            return 0;
        }
        if (set.contains(root.val)) {
            set.remove(root.val);
        }
        else {
            set.add(root.val);
        }
        if(root.left==null&& root.right==null){
            return set.size()<=1?1:0; 
        }
        return dfs(root.left, new HashSet<>(set)) + dfs(root.right, new HashSet<>(set));
    }
}

File: my-folder/1568-pseudo-palindromic-paths-in-a-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pseudoPalindromicPaths (self, root: Optional[TreeNode]) -> int:
        def dfs(root, st):
            if not root:
                return 0
            if root.val in st:
                st.remove(root.val)
            else:
                st.add(root.val)
            if not root.left and not root.right:
                if len(st) <= 1:
                    return 1
                else:
                    return 0
            return dfs(root.left, st.copy()) + dfs(root.right, st.copy())


        return dfs(root, set())
